rust/xaynet-analytics/src/data_combiner/data_combiner.rs

```diff
@@ -0,0 +1,235 @@
+use chrono::{DateTime, Datelike, Duration, NaiveDate, Utc};
+use std::time::UNIX_EPOCH;
+
+use crate::data_combiner::data_points::data_point::{
+    Calculate, DataPoint, DataPointMetadata, Period, PeriodUnit,
+};
+use crate::data_combiner::data_points::screen_active_time::ScreenActiveTime;
+use crate::data_combiner::data_points::screen_enter_count::ScreenEnterCount;
+use crate::data_combiner::data_points::was_active_each_past_period::WasActiveEachPastPeriod;
+use crate::data_combiner::data_points::was_active_past_n_days::WasActivePastNDays;
+use crate::repo::analytics_event::AnalyticsEvent;
+use crate::repo::repo::Repository;
+
+pub struct DataCombiner<R> {
+    repo: Box<R>,
+}
+
+impl<R> DataCombiner<R>
+where
+    R: Repository,
+{
+    pub fn new(repo: Box<impl Repository>) -> DataCombiner<impl Repository> {
+        DataCombiner { repo }
+    }
+
+    pub fn init_data_points(&self) -> Vec<DataPoint> {
+        [
+            self.to_data_points(self.init_was_active_past_n_days_vars()),
+            self.to_data_points(self.init_screen_active_time_vars()),
+            self.to_data_points(self.init_screen_enter_count_vars()),
+            self.to_data_points(self.init_was_active_each_past_period_vars()),
+        ]
+        .concat()
+    }
+
+    fn to_data_points(&self, calculables: Vec<impl Calculate>) -> Vec<DataPoint> {
+        calculables
+            .into_iter()
+            .map(|calculable| DataPoint::new(calculable.metadata(), calculable.calculate()))
+            .collect()
+    }
+
+    fn init_was_active_past_n_days_vars(&self) -> Vec<impl Calculate> {
+        let metadatas = vec![
+            DataPointMetadata {
+                period: Period {
+                    unit: PeriodUnit::Days,
+                    n: 1,
+                },
+                end: Utc::now(),
+            },
```

Comment:
is there a way to instantiate these `DataPointMetadata`s on a single line? it looked so much nicer before the formatter did its job

Reply:
thanks a lot! will do